server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/verumcoin/VerumcoinExtension.java

```diff
@@ -0,0 +1,128 @@
+import com.generalbytes.batm.server.extensions.*;
+import com.generalbytes.batm.server.extensions.FixPriceRateSource;
+import com.generalbytes.batm.server.extensions.ExtensionsUtil;
+import com.generalbytes.batm.server.extensions.extra.verumcoin.wallets.verumcoind.VerumcoindRPCWallet;
+import com.generalbytes.batm.server.extensions.extra.verumcoin.wallets.verumcoind.VerumcoindUniqueAddressRPCWallet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+import java.net.InetSocketAddress;
+import java.util.*;
+
+public class VerumcoinExtension extends AbstractExtension{
+    private static final Logger log = LoggerFactory.getLogger(VerumcoinExtension.class);
+
+    @Override
+    public String getName() {
+        return "BATM Verumcoin extension";
+    }
+
+    @Override
+    public IWallet createWallet(String walletLogin, String tunnelPassword) {
+        try {
+        if (walletLogin !=null && !walletLogin.trim().isEmpty()) {
+            StringTokenizer st = new StringTokenizer(walletLogin,":");
+            String walletType = st.nextToken();
+
+            if ("verumcoind".equalsIgnoreCase(walletType)
+                || "verumcoindnoforward".equalsIgnoreCase(walletType)) {
+                //"verumcoind:protocol:user:password:ip:port:accountname"
+
+                String protocol = st.nextToken();
+                String username = st.nextToken();
+                String password = st.nextToken();
+                String hostname = st.nextToken();
+                int port = Integer.parseInt(st.nextToken());
+                String label = "";
+                if (st.hasMoreTokens()) {
+                    label = st.nextToken();
+                }
+
+                InetSocketAddress tunnelAddress = ctx.getTunnelManager().connectIfNeeded(walletLogin, tunnelPassword, InetSocketAddress.createUnresolved(hostname, port));
+                hostname = tunnelAddress.getHostString();
+                port = tunnelAddress.getPort();
+
+                if (protocol != null && username != null && password != null && hostname !=null && label != null) {
+                    String rpcURL = protocol +"://" + username +":" + password + "@" + hostname +":" + port;
+                    if ("verumcoindnoforward".equalsIgnoreCase(walletType)) {
+                        return new VerumcoindUniqueAddressRPCWallet(rpcURL);
+                    }
+                    return new VerumcoindRPCWallet(rpcURL, label);
+                }
+            }
+        }
+        } catch (Exception e) {
+            log.warn("createWallet failed for prefix: {}, {}: {} ",
+                ExtensionsUtil.getPrefixWithCountOfParameters(walletLogin), e.getClass().getSimpleName(), e.getMessage()
+            );
+        }
+        return null;
+    }
+
+    @Override
+    public ICryptoAddressValidator createAddressValidator(String cryptoCurrency) {
+        if (CryptoCurrency.LTC.getCode().equalsIgnoreCase(cryptoCurrency)) {
```

Comment:
Should be VERUM instead of LTC? please double check for copy paste errors in the rest of the code too

Reply:
Hello, thanks for help! We`ve updated PR with a fix